src/components/dds/textInput/TextInput.tsx

```diff
@@ -0,0 +1,148 @@
+import { uniqueId } from 'lodash';
+import getRootClassNames from '../common/getRootClassNames';
+import * as React from 'react';
+import classNames from 'classnames';
+import type { Type } from 'dds-core/components/text-input/ts/input';
+import { componentClasses as inputClasses } from 'dds-core/components/text-input/ts/input';
+import { componentClasses as labelClasses } from 'dds-core/components/label/label';
+import { componentClasses as passwordClasses, defaultProps } from 'dds-core/components/input-password/input-password';
+import 'dds-core/components/text-input/_text-input.scss';
+import 'dds-core/components/label/_label.scss';
+import 'dds-core/components/input-password/_input-password.scss';
+import 'dds-core/components/form/_form.scss';
+import type { Icons, IconType } from 'dds-core/components/icon/ts/icon';
+import { useToggle } from 'hooks';
+import { Icon } from '../icon';
+
+export interface TextInputProps extends React.InputHTMLAttributes<HTMLInputElement> {
+    /**
+     * 	The input type to use with this component (text|url|search|tel|email)
+     */
+    type?: Type;
+    /**
+     * 	The name of a DDS icon (without the prefix *dds__*)
+     */
+    iconName?: Icons;
+    /**
+     * Define the type of icon that will be added (font-icon|svg)
+     */
+    iconType?: IconType;
+    /**
+     * Whether the text input should be disabled
+     */
+    inactive?: boolean;
+    /**
+     * Toggles the 'invalid' attribute. Used when the component is in a form with validation.
+     * */
+    invalid?: boolean;
+    /**
+     * Label text for the text input
+     * */
+    labelText?: string;
+    /**
+     * Minimum length of the text input
+     * */
+    minLength?: number;
+    /**
+     * Maximum length of the text input
+     * */
+    maxLength?: number;
+    /**
+     * Name of the text input
+     * */
+    name?: string;
+    /**
+     * 	Provides a brief hint to the user as to what kind of information is expected in the field
+     * */
+    placeholder?: string;
+    /**
+     * 	Toggles the 'required' attribute. Used when the component is in a form with validation.
+     */
+    required?: boolean;
+    /**
+     * Error text for the text input
+     * */
+    errorText?: string;
+    /**
+     * Helper text for the text input
+     * */
+    helperText?: string;
+    /**
+     * Specifies the initial value for this component
+     * */
+    value?: string;
+    /**
+     * Function called on input value change
+     */
+    onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
```

Comment:
For ease of use of this component, how about:
```suggestion
    onChange?: (value: string) => void;
```
or
```suggestion
    onChange?: (value: string, event: React.ChangeEvent<HTMLInputElement>) => void;
```
?
I think every usage of this callback will be extracting the value from the event, so it would be beneficial to receive the value directly.

Alternatively we could stick to signature specific to Semantic UI, as we already have `useInput` and `useInputs` hooks that work with that:
```suggestion
    onChange?: (event: React.ChangeEvent<HTMLInputElement>, data: InputOnChangeData) => void;
```
(however I'm not a fan of such arguments order, I'd prefer to have `data` first, as it is frequently used, and `event` second, as it is frequently ignored)

Reply:
changed it to second proposition: https://github.com/cloudify-cosmo/cloudify-ui-components/pull/219/commits/e2d1045b44dce89e23b4eb2e758475ec89d4ec48